### PR TITLE
Finish support for RFC4180 for CSV bulk insert operations

### PR DESCRIPTION
**Pull request by funkyjive**
RFC4180 specifies support for fields that contain newlines as long as they are quoted.  See section 6 in the specification here: https://www.ietf.org/rfc/rfc4180.txt

This PR enhances the read operation when the option escapeColumnDelimitersCSV is set to true to allow newline characters to survive the CSV parse process and find their way into the record.  An existing unit test was enhanced to test the behavior.

### CODE REVIEW

**Comment by funkyjive** [comment]
> @funkyjive please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement
@microsoft-github-policy-service agree

**Comment by lilgreenbird** [comment]
thank you for your contribution, the team will take a look at this when time permits

**Review by barryw-mssql — DISMISSED** [review]
Appears to handle newline in Mac, Linux, and Windows formats

**Review by Jeffery-Wasty — DISMISSED** [review]

File: src/test/resources/BulkCopyCSVTestInputDelimiterEscape.csv [hunk 1]
```diff
@@ -9,3 +9,7 @@
 9@1997@Ford@E350@"Super@ ""luxurious"" truck"@ 
 10@1997@ "Ford" @E350@  "E63"@ 
 11@1997@Ford@E350@" Super luxurious truck "@ 
+12@1997@"F
+o
+r
+d"@"E350"@"""Super"" ""luxurious"" ""truck"""@
```
Comment (lilgreenbird) [inline src/test/resources/BulkCopyCSVTestInputDelimiterEscape.csv:15]: needs new line here

Comment (funkyjive) [reply src/test/resources/BulkCopyCSVTestInputDelimiterEscape.csv:15]: Where would you like it?  Are you wanting to break up the "Super" "luxurious" "truck" field?

Comment (lilgreenbird) [reply src/test/resources/BulkCopyCSVTestInputDelimiterEscape.csv:15]: I mean at the very end of the file where you see the red mark

**Review by tkyc — APPROVED** [review]

**Review by lilgreenbird — CHANGES_REQUESTED** [review]
please use file formatter to format this PR all files need to be formatted before merge

File: src/test/resources/BulkCopyCSVTestInputDelimiterEscape.csv (hunk 1 above)
Comment (funkyjive) [reply src/test/resources/BulkCopyCSVTestInputDelimiterEscape.csv:15]: Ok, sure.

**Comment by funkyjive** [comment]
> please use file formatter to format this PR all files need to be formatted before merge

I think I have what you are looking for now.

**Review by lilgreenbird — APPROVED** [review]

**Review by Jeffery-Wasty — APPROVED** [review]

**Review by barryw-mssql — APPROVED** [review]

**Review by tkyc — APPROVED** [review]

**Pull request merged**